File: for-local-test/145.binary-tree-postorder-traversal.java
import java.util.ArrayList;
import java.util.List;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        if(root == null) return res;
        postorder(root, res);
        return res;
    }
    private void postorder(TreeNode root, List<Integer> res){
        if(root == null) return;
        postorder(root.left, res);
        postorder(root.right, res);
        res.add(root.val);
    }

    public static void main(String[] args){
        TreeNode root = new TreeNode(1);
        root.right = new TreeNode(2);
        root.right.left = new TreeNode(3);
        Solution sol = new Solution();
        List<Integer> res = sol.postorderTraversal(root);
        System.out.println(res);
    }
}

File: for-local-test/python/145.binary-tree-postorder-traversal.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def postorderTraversal(self, root) -> list[int]:
        res = []
        self._postorder(root, res)
        return res
    def _postorder(self, root, res):
        if not root:
            return
        self._postorder(root.left, res)
        self._postorder(root.right, res)
        res.append(root.val)
    
if __name__ == "__main__":
    # Create a binary tree
    root = TreeNode(1)
    root.right = TreeNode(2)
    root.right.left = TreeNode(3)

    solution = Solution()
    print(solution.postorderTraversal(root))  # Output: [3, 2, 1]